src/corecel/math/Atomics.hh

```diff
@@ -28,7 +28,7 @@ CELER_FORCEINLINE_FUNCTION T atomic_add(T* address, T value)
 #else
     CELER_EXPECT(address);
     T initial;
-#    ifdef _OPENMP
+#    if CELERITAS_OPENMP == CELERITAS_OPENMP_TRACK
```

Comment:
Do we need to keep the `#ifdef _OPENMP` to suppress the unknown pragmas warnings?
```
                 from /home/alund/celeritas_project/celeritas/test/corecel/data/StackAllocator.test.cc:8:
/home/alund/celeritas_project/celeritas/src/corecel/math/Atomics.hh:33: warning: ignoring ‘#pragma omp atomic’ [-Wunknown-pragmas]
   33 | #        pragma omp atomic capture
      | 
/home/alund/celeritas_project/celeritas/src/corecel/math/Atomics.hh:83: warning: ignoring ‘#pragma omp atomic’ [-Wunknown-pragmas]
   83 | #        pragma omp atomic capture
      | 
/home/alund/celeritas_project/celeritas/src/corecel/math/Atomics.hh:106: warning: ignoring ‘#pragma omp atomic’ [-Wunknown-pragmas]
  106 | #        pragma omp atomic capture
...
```

Reply:
Hmmm actually this points out that we're only "atomic" if OpenMP is enabled for a translation unit... that's a little surprising. If we forget to link the OpenMP target in one module and access the "atomic" data it'll be non-atomic. I can't think where this would happen in real life though...